src/node_options.cc

```diff
@@ -361,9 +375,13 @@ EnvironmentOptionsParser::EnvironmentOptionsParser() {
             "set module type for string input",
             &EnvironmentOptions::module_type,
             kAllowedInEnvironment);
-  AddOption("--es-module-specifier-resolution",
+  AddOption("--experimental-specifier-resolution",
             "Select extension resolution algorithm for es modules; "
             "either 'explicit' (default) or 'node'",
+            &EnvironmentOptions::experimental_specifier_resolution,
+            kAllowedInEnvironment);
+  AddOption("--es-module-specifier-resolution",
+            "",
```

Comment:
Fun fact, leaving no description on a flag stop it from printing when you run `node -h` MAGIC!

Reply:
We should probably adopt the same for the aliasing of `--experimental-loader` and `--loader` actually.

Reply:
If this is an alias, shouldn't it be using [`AddAlias()`](https://github.com/nodejs/node/blob/6bf5a1d691291cdfcc4941e68f00d0003e565476/src/node_options.h#L309-L319) rather than `AddOption()`?

Reply:
I originally implemented it that way but couldn't figure out how to enforce behavior that both the alias and the actual command couldn't be used at the same time

Ended up manually implemented this [here](https://github.com/nodejs/node/pull/30678/files#diff-36016d459a1aea31eea2bfe332750c1aR131-R149).

There was weird behavior depending on the order of arguments, so I opted to do this instead of hunting down how to get the same behavior as an alias. Is there a better way to do it?

Reply:
> I originally implemented it that way but couldn't figure out how to enforce behavior that both the alias and the actual command couldn't be used at the same time
> 
> Ended up manually implemented this [here](https://github.com/nodejs/node/pull/30678/files#diff-36016d459a1aea31eea2bfe332750c1aR131-R149).
> 
> There was weird behavior depending on the order of arguments, so I opted to do this instead of hunting down how to get the same behavior as an alias. Is there a better way to do it?

¯\\_(ツ)\_/¯ cc @addaleax

Reply:
> I originally implemented it that way but couldn't figure out how to enforce behavior that both the alias and the actual command couldn't be used at the same time

I guess we could add a special kind of alias for this, but … why?

Reply:
Alternatively, I think this could use at least `Implies()` to clarify the relation between the two options

Reply:
If y'all are happy with just shipping this as is, it is a one off and something that will eventually get removed.